docs/contributing/development-guide.md

```diff
@@ -67,6 +67,26 @@ $ pip install -r requirements.txt
 $ pip install -e .
 ```
 
+## Architecture Overview
+
+![Stravalib Architecture](../images/stravalib_architecture.png)
+
+Stravalib contains the following main components:
+
+At the core, a (pydantic) domain model is generated and updated by a bot via pull requests.
+This model reflects the officially published API specification by Strava and is stored in
+the module `strava_model.py`. This file should never be edited other than by the pull requests
+generated by the bot.
+
+For allowing custom typing, unit conversion, (de-)serialization behavior, and support of
+undocumented Strava features, the module `model.py` contains classes that inherit from the
+official Strava domain model in `strava_model.py`.
```

Comment:
```suggestion
The module `model.py` contains classes that inherit from the
official Strava domain model in `strava_model.py`. This module, supports custom typing, unit conversion, (de-)serialization behavior, and support for
undocumented Strava features, .
```

not sure if this is what you mean just found this sentence a bit confusing as written.